docs/DeveloperGuide.md

```diff
@@ -239,6 +239,40 @@ _{more aspects and alternatives to be added}_
 _{Explain here how the data archiving feature will be implemented}_
 
 
+### Finding students
+
+The `FindCommand` allows users to find students based on person fields.
+
+**Current Implementation**
+
+The figure below shows the sequence diagram for finding students.
+
+![Find Command](images/FindSequenceDiagram.png)
+
+
+The user can specify multiple fields to search for and each field takes in multiple keywords.
+An exception is the tag field which only accepts a single keyword. Instead, users can specify multiple tag fields.
+
+The `PersonMatchesKeywordsPredicate` is used to test whether a person matches the specified keywords.
+For each searchable field, a predicate is created which tests if a given person's field contains all specified keywords.
+
+Additionally, a `FindCondition` can be specified by the user which determines whether `all`, `any` or `none` of the fields specified are required to match with a given person, for the person to be returned. The default is `all`.
+
+The `PersonMatchesKeywordsPredicate#test()` method will compose all searched field predicates into a single predicate, depending on the find condition.
+The predicate is then used to filter the list of person.
+
+
+**Design considerations:**
+
+**Aspect: Data structure of predicates**
+* **Alternative 1 (current choice):** Use a single `PersonMatchesKeywordsPredicate` class to represent all fields' predicates.
+    * Pros: All predicates meant for testing on person fields can be consolidated in one place and reduce code duplication. It can also handle encapsulate the composition of all predicates.
```

Comment:
```suggestion
    * Pros: All predicates meant for testing on person fields can be consolidated in one place and reduce code duplication. It can also encapsulate the composition of all predicates.
```